language/move-lang/stdlib/modules/vector.move

```diff
@@ -56,4 +55,35 @@ module Vector {
         length(v) == 0
     }
 
+    // Return true if `e` is in the vector `v`
+    public contains<Element>(v: &T<Element>, e: &Element): bool {
+        let i = 0;
+        let len = length(v);
+        while (i < len) {
+            if (borrow(v, i) == e) return true;
+            i = i + 1;
+        };
+        false
+    }
+
+    // Remove the `i`th element E of the vector, shifting all subsequent elements
+    // It is O(n) and preserves ordering
+    public remove<Element>(v: &mut T<Element>, i: u64): Element {
+        let len = length(v);
+        // i out of bounds; abort
+        if (i >= len) abort 10;
+
+        len = len - 1;
+        while (i < len) swap(v, i, { i = i + 1; i });
+        pop_back(v)
+    }
+
+    // Remove the `i`th element E of the vector by swapping it with the last element,
+    // and then popping it off
+    // It is O(1), but does not preserve ordering
+    public swap_remove<Element>(v: &mut T<Element>, i: u64): Element {
```

Comment:
Looks much nicer in the source lang!

Will the borrow checker complain if you try to inline `last_idx` here?

Reply:
It's a bit weird, and I'll need to talk to @shazqadeer about it. In short, I think we are doing some non optimal things with regards to borrow local and copy local